src/Microsoft.AspNet.Hosting/Internal/HostingApplicationFeature.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    public class HostingApplicationFeature
+    {
+        public HostingApplicationFeature(IDisposable logScope, int startTime)
```

Comment:
See https://github.com/aspnet/Hosting/commit/dfae316be07d944433a013a181bc2b01837f55e5 for clarification about what the earlier note meant

Adding a feature to the http context would add more overhead, the approach in the example doesn't allocate any heap objects or add anything to feature dictionaries